### DISCUSSION TITLE
Remeshing a function / changing mesh coordinates while preserving function shape

### DISCUSSION THREAD

**Discussion by sorenandflying** [firedrakeproject/firedrake#3101] [Firedrake support]
Hi All,

I am working on a project where I need to change the coordinates of a mesh so I am working with a smaller and smaller section of a function.

While I can [change mesh coordinates](https://www.firedrakeproject.org/mesh-coordinates.html), this ends up "squishing" my function, rather than "cutting" or "zooming" into the smaller section I want to work with.

I can evaluate individual points of a function `f_0` using `at()`, but I need to evaluate a whole sections of `f_0` using the same number of cells and the same degree as the original function. I hope my code below makes it a little more clear what I am attempting.

Here is an example of code of some of the variations I've tried so far.

```python
import firedrake
import matplotlib.pyplot as plt
import numpy as np

# Initialize test mesh
Lx_0 = 10
nx = 10
deg = 2

mesh_0 = firedrake.IntervalMesh(nx, Lx_0)
Q_0 = firedrake.FunctionSpace(mesh_0, "CG", deg)

# Define simple function
from firedrake import sin
x_0, = firedrake.SpatialCoordinate(mesh_0)
f_0 = firedrake.interpolate(sin(x_0*3.14159),Q_0);

# Display function
fig, axes = plt.subplots()
firedrake.plot(f_0, axes = axes);
```
<img width="569" alt="Screenshot 2023-09-14 at 12 41 20 AM" src="https://github.com/firedrakeproject/firedrake/assets/39568390/bd610c7f-dd1c-492c-a7ac-a0309267afcd">

```python
# Define specs of new mesh and function space
Lx_1 = 5
mesh_1 = firedrake.IntervalMesh(nx, Lx_0)
Q_1 = firedrake.FunctionSpace(mesh_1, "CG", deg)
```

```python
# Interpolating to a new mesh throws an error
f_1 = firedrake.interpolate(f_0, Q_1)

# Evaluating the function at mesh points throws an error
x_2, = firedrake.SpatialCoordinate(mesh_1)
f_2 = firedrake.interpolate(f_0.at(x_2), Q_1)
```

```python
# Directly altering mesh changes coordinates, but squishes function after interpolation
mesh_0.coordinates.dat.data[:] *= Lx_1/Lx_0
f_2 = firedrake.interpolate(f_0, Q_0)

fig, axes = plt.subplots()
firedrake.plot(f_2, axes = axes);
```
<img width="574" alt="Screenshot 2023-09-14 at 12 46 18 AM" src="https://github.com/firedrakeproject/firedrake/assets/39568390/5ed3389b-518f-4230-8d66-db95cc9772d2">

The result I want looks like the following chart, but I need it to be based on the values of 'f_0`:
<img width="560" alt="Screenshot 2023-09-14 at 12 48 05 AM" src="https://github.com/firedrakeproject/firedrake/assets/39568390/09e81704-8d4c-46f5-bee9-323058a1305a">

Thanks!

**Comment by connorjward** [comment]
Have you recently run `firedrake-update`? As of very recently we now support [mesh-to-mesh interpolation](https://www.firedrakeproject.org/interpolation.html#interpolation-across-meshes).
